lib/active_merchant/billing/gateways/place_to_pay.rb

```diff
@@ -0,0 +1,196 @@
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_auth(post, options)
+        post[:internalReference] = options[:internalReference]
+        post[:authorization] = authorization                
+        post[:action] = 'reverse' #Allowed values: reverse refund process void dispersion pre_authorization checkin checkout reauthorization 
+        commit(:post, 'gateway/transaction', post)
+      end
+
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((tranKey)\w+), '\1[FILTERED]').
+          gsub(/(number\\?":\\?")(\d*)/, '\1[FILTERED]').
+          gsub(/(cvv\\?":\\?")(\d*)/, '\1[FILTERED]')
+      end
+
+      private
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def build_url(action, base)
+        url = base
+        url += action
+        url
+      end
+
+      def commit(method, action, parameters)
+        base_url = (test? ? test_url : live_url)
+        url = build_url(action, base_url)
+
+        begin
+          url = (test? ? test_url : live_url) + action.to_s
+          rel_path = "#{method}/v1/#{action}"
+          response = api_request(method, url, rel_path, parameters)
+
+        rescue ResponseError => e
+          raw_response = e.response.body
+          response = parse(raw_response)
+        end
+
+        Response.new(
+          success_from(action, response, options),
+          message_from(response),
+          response,
+          test: test?,
+          error_code: error_code_from(action, response, options),
+          authorization: response['authorization'],
+          network_transaction_id: response['internalReference'],
+        )
+      end
+
+
+      def success_from(action, response, options)
+        case action
+        when :authonly.to_s, :lookup.to_s, :mpiQuery.to_s, :interests.to_s, :generate_otp.to_s,
+          :validate_otp.to_s, :capture.to_s, :search.to_s
+          return response['status'] && response['status']['status'] === RESPONSES_STATUS[:ok]
+        when :sale.to_s, :query_transaction.to_s, :refund.to_s
+          return response['status'] && response['status']['status'] === RESPONSES_STATUS[:approved]
+        else
+          return response['status']['status'];
+        end
+      end
+
+      def message_from(response)
+        response['status'] && response['status']['message'] ? response['status']['message'] :
+          COMMON_MESSAGES[:missed]
+      end
+
+      def authorization_from(response); end
+
+      def post_data(action, parameters = {})
+        JSON.generate(parameters)
+      end
+
+      def error_code_from(action, response, options)
+        unless success_from(action, response, options)
+          # TODO: lookup error code for this response
+          STANDARD_ERROR_CODE_MAPPING[response['reason']]
```

Comment:
Is `STANDARD_ERROR_CODE_MAPPING` used? It's an empty hash at the top of the file.

Reply:
Now we are using it. We mapped some responses